### PR TITLE
site: Add command-line download instructions for windows binary

### LINKED ISSUE #11861: site: Add command-line download instructions for windows binary
how about as a follow up PR fix the windows "Download" so we have a command to download binaries for windows  too ? instead of asking peoople to go to the release page
https://github.com/medyagh/minikube/blob/b5cf45a1d47f1660a7e91075269000d4d0c29aa6/site/content/en/docs/start/_index.md#L211

something like this might work ? (that we use to download gopogh)
```
 (New-Object Net.WebClient).DownloadFile("https://github.com/medyagh/gopogh/releases/download/v0.8.0/gopogh.exe", "C:\ProgramData\chocolatey\bin\gopogh.exe")
```

_Originally posted by @medyagh in https://github.com/kubernetes/minikube/pull/11859#issuecomment-872505225_

### PR DESCRIPTION
**Pull request by mahalrs**
fixes #11861

https://deploy-preview-11872--kubernetes-sigs-minikube.netlify.app/docs/start/

---

Screen Shot 2021-07-07 at 10 14 49 PM

Screen Shot 2021-07-07 at 10 14 57 PM

### CODE REVIEW

**Comment by mahalrs** [comment]
@medyagh let me know if this looks good or you have any suggestions.

/assign @medyagh
/kind documentation
/priority important-longterm

**Comment by medyagh** [comment]
@mahalrs I wonder if we can come up with a universal place in the path ? Or help user to add a standard place in path and have minikube add to that place ? This would help us with the support and upgrading later

**Comment by mahalrs** [comment]
> @mahalrs I wonder if we can come up with a universal place in the path ? Or help user to add a standard place in path and have minikube add to that place ? This would help us with the support and upgrading later

What about creating a separate folder: `C:\minikube\minikube.exe` And then providing step-by-step instructions for adding that folder to path.

**Comment by medyagh** [comment]
> > @mahalrs I wonder if we can come up with a universal place in the path ? Or help user to add a standard place in path and have minikube add to that place ? This would help us with the support and upgrading later
> 
> How about creating a separate folder: `C:\minikube\minikube.exe` And then providing step-by-step instructions for adding that folder to path.

That sounds reasonable. We could add instructions to create dirs

**Comment by medyagh** [comment]
> > > @mahalrs I wonder if we can come up with a universal place in the path ? Or help user to add a standard place in path and have minikube add to that place ? This would help us with the support and upgrading later
> > 
> > 
> > How about creating a separate folder: `C:\minikube\minikube.exe` And then providing step-by-step instructions for adding that folder to path.
> 
> That sounds reasonable. We could add instructions to create dirs

Or we could follow kubectl's way... Where does kubectl install their binary in windows ? We could be in the same folder as them

**Comment by mahalrs** [comment]
> > > > @mahalrs I wonder if we can come up with a universal place in the path ? Or help user to add a standard place in path and have minikube add to that place ? This would help us with the support and upgrading later
> > > 
> > > 
> > > How about creating a separate folder: `C:\minikube\minikube.exe` And then providing step-by-step instructions for adding that folder to path.
> > 
> > 
> > That sounds reasonable. We could add instructions to create dirs
> 
> Or we could follow kubectl's way... Where does kubectl install their binary in windows ? We could be in the same folder as them

kubectl's installation page has a step "Add the binary in to your `PATH`" but it doesn't mention any particular location to install it or any steps to setup the `PATH` (I guess they left it to the user to choose the location and properly setup the path) https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/. Should we follow kubectl's way?

**Comment by medyagh** [comment]
> kubectl's installation page has a step "Add the binary in to your `PATH`" but it doesn't mention any particular location to install it or any steps to setup the `PATH` (I guess they left it to the user to choose the location and properly setup the path) https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/. Should we follow kubectl's way?

ah ! I rather we provide a set of commands users can copy/paste and not figure out their path !
(if we can make it in a way that using commands, we update their path to C:\minikube\ 
then I am fine with that !)

**Comment by mahalrs** [comment]
> > kubectl's installation page has a step "Add the binary in to your `PATH`" but it doesn't mention any particular location to install it or any steps to setup the `PATH` (I guess they left it to the user to choose the location and properly setup the path) https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/. Should we follow kubectl's way?
> 
> ah ! thats a bummer, that doesnt sound like a smooth experience for windows users, I rather we provide a set of commands users can copy/paste and not figure out their path !
> (if we can make it in a way that using commands, we update their path to C:\minikube\
> then I am fine with that !)

Now you can download the Windows binary by just clicking the link, no need to go to the release page. Also, added instructions to setup the `PATH` (it seems like there isn't any simple way to setup `PATH` using the command line). @medyagh let me know if this looks good or you have any suggestions.

File: site/content/en/docs/start/_index.md [hunk 1]
```diff
@@ -423,15 +423,63 @@ choco install minikube
...
+1. Download the <a href="#" id="latest-beta-download-link">latest beta release</a>.  
+<br>
+    Or if you have `curl` installed, use this command:
+    ```shell
+    r=https://api.github.com/repos/kubernetes/minikube/releases
+    curl -Lo minikube.exe $(curl -s $r | grep -o 'http.*download/v.*beta.*/minikube-windows-amd64.exe' | head -n1)
+    ```
+
+2. Add the binary in to your `PATH`.
+    - Create a new folder in C drive, name it `minikube` (recommended) or whatever you like.
```
Comment (medyagh) [inline site/content/en/docs/start/_index.md:458]: this seems to be too wordy, I prefer all of these to be down by copy pasting commands.

File: site/content/en/docs/start/_index.md [hunk 2]
```diff
@@ -423,15 +423,63 @@ choco install minikube
...
 
 {{% quiz_instruction id="/Windows/x86-64/Beta/.exe download" %}}
-Download and run the stand-alone minikube Windows installer from [the release page](https://github.com/kubernetes/minikube/releases).
-
-_If you used a CLI to perform the installation, you will need to close that CLI and open a new one before proceeding._
+1. Download the <a href="#" id="latest-beta-download-link">latest beta release</a>.  
+<br>
+    Or if you have `curl` installed, use this command:
+    ```shell
+    r=https://api.github.com/repos/kubernetes/minikube/releases
```
Comment (medyagh) [inline site/content/en/docs/start/_index.md:453]: I am curious have you verified this yourself on a windows machine?

Comment (mahalrs) [reply site/content/en/docs/start/_index.md]: Thanks for pointing this out. It took a while to learn the basics of PowerShell but now I have updated the instructions to set path using the command-line, and have tested everything on Windows.

File: site/content/en/docs/start/_index.md (hunk 1 above)
Comment (mahalrs) [reply site/content/en/docs/start/_index.md]: Fixed.

File: site/content/en/docs/start/_index.md [hunk 3]
```diff
@@ -423,15 +423,62 @@ choco install minikube
...
+    if($oldpath -notlike "*;C:\minikube*"){`
+      [Environment]::SetEnvironmentVariable("Path", $oldpath+";C:\minikube", [EnvironmentVariableTarget]::Machine)`
+    }
+    ```
+    _If you used a CLI to perform the installation, you will need to close that CLI and open a new one before proceeding._
+<script type="text/javascript">
+  fetch("https://api.github.com/repos/kubernetes/minikube/releases")
+    .then(response => response.text())
+    .then(data => {
+      let u = data.match(/http.*download\/v.*beta.*\/minikube-installer\.exe/)[0]
```
Comment (andriyDev) [inline site/content/en/docs/start/_index.md:473]: const instead of let.

File: site/content/en/docs/start/_index.md [hunk 4]
```diff
@@ -423,15 +423,62 @@ choco install minikube
...
+      document.getElementById("latest-beta-download-link").href=u;
+    })
+    .catch((error) => {
+      let el = document.getElementById("latest-beta-download-link");
```
Comment (andriyDev) [inline site/content/en/docs/start/_index.md:477]: const instead of let.

File: site/content/en/docs/start/_index.md [hunk 5]
```diff
@@ -423,15 +423,62 @@ choco install minikube
...
+      [Environment]::SetEnvironmentVariable("Path", $oldpath+";C:\minikube", [EnvironmentVariableTarget]::Machine)`
+    }
+    ```
+    _If you used a CLI to perform the installation, you will need to close that CLI and open a new one before proceeding._
+<script type="text/javascript">
+  fetch("https://api.github.com/repos/kubernetes/minikube/releases")
+    .then(response => response.text())
+    .then(data => {
+      let u = data.match(/http.*download\/v.*beta.*\/minikube-installer\.exe/)[0]
+      document.getElementById("latest-beta-download-link").href=u;
```
Comment (andriyDev) [inline site/content/en/docs/start/_index.md:474]: spaces around equal.

**Review by andriyDev — CHANGES_REQUESTED** [review]
These changes are very pedantic, sorry :P

File: site/content/en/docs/start/_index.md (hunk 3 above)
Comment (mahalrs) [reply site/content/en/docs/start/_index.md]: done

**Comment by mahalrs** [comment]
@andriyDev I have made the requested changes. Let me know if this looks good!

**Comment by andriyDev** [comment]
/approve

Also getting @medyagh to rereview it! Sorry for the delay!

**Review by medyagh — APPROVED** [review]

**Comment by medyagh** [comment]
thank you @mahalrs this will improve the experience for windows users's installing experience
